addons/html_builder/static/src/builder/plugins/visibility_plugin.js

```diff
@@ -0,0 +1,64 @@
+import { Plugin } from "@html_editor/plugin";
+import { isMobileView } from "@html_builder/builder/utils/utils";
+
+export class VisibilityPlugin extends Plugin {
+    static id = "visibility";
+    resources = {
+        on_target_show: this.onTargetShow.bind(this),
+        on_target_hide: this.onTargetHide.bind(this),
+    };
+
+    onTargetHide(snippetEl) {
+        switch (getSelectorName(snippetEl)) {
+            case "DeviceVisibility": {
+                hideInvisibleEl(snippetEl);
+                return;
+            }
+            case "ConditionalVisibility": {
+                hideInvisibleEl(snippetEl);
+                if (snippetEl.classList.contains("o_snippet_invisible")) {
+                    snippetEl.classList.add("o_conditional_hidden");
+                }
+                return;
+            }
+        }
+    }
+    onTargetShow(snippetEl) {
+        switch (getSelectorName(snippetEl)) {
+            case "DeviceVisibility": {
+                showInvisibleEl(snippetEl);
+                return;
+            }
+            case "ConditionalVisibility": {
+                showInvisibleEl(snippetEl);
+                snippetEl.classList.remove("o_conditional_hidden");
+                return;
+            }
+        }
+    }
+}
```

Comment:
I think that instead of making cases about the different selectors and all, you should add the `on_target_show/hide` resources directly in the different option plugins :thinking: 
It will allow to have all the option specific behavior/logic at one place and the dispatches will just call all of them and each of them will know what to do based on its selector :slightly_smiling_face: 

That's what we are planning to do with the `on_move/on_remove`, so I think these ones should be like this too :slightly_smiling_face:
You can extract the common logic where the dispatch is called I guess (for the duplicated code)